storage/src/main/java/tech/pegasys/artemis/storage/client/CombinedChainDataClient.java

```diff
@@ -233,6 +233,16 @@ public boolean isFinalizedEpoch(final UnsignedLong epoch) {
     return getBestBlockRoot().map(store::getBlockState);
   }
 
+  public Optional<StateAndBlockRoot> getHeadStateAndBlockRootFromStore() {
```

Comment:
This feels like it might be overkill and it would be better for callers to just get the best block root and then get the state from that as they need it.  There's quite a lot of methods like this starting to pop up which individually are convenient but collectively make it complex to know which one you should call.

Reply:
Refactored accordingly.